cores/arduino/HardwareSerial.h

```diff
@@ -112,6 +112,8 @@ class HardwareSerial : public Stream {
   public:
     HardwareSerial(uint32_t _rx, uint32_t _tx);
     HardwareSerial(PinName _rx, PinName _tx);
+    HardwareSerial(uint32_t _rx, uint32_t _tx, uint32_t _rts, uint32_t _cts);
+    HardwareSerial(PinName _rx, PinName _tx, PinName _rts, PinName _cts);
```

Comment:
Maybe we could simplify like this to avoid declare new constructor?

```suggestion
    HardwareSerial(uint32_t _rx, uint32_t _tx, , uint32_t _rts = NUM_DIGITAL_PINS, uint32_t _cts = NUM_DIGITAL_PINS);
    HardwareSerial(PinName _rx, PinName _tx, , PinName _rts = NC, PinName _cts = NC);
```

Reply:
I want to avoid 3-parameters call (rx, tx, rts) of constructor that is a bit ambiguous usage.
But it's not a major problem, and someone feels it convenient to only use RTS flow control.
So I can follow your suggestion.

Reply:
I reduced repeated constructors and it looks prettier. Thank you @fpistm.